Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1284,6 +1284,23 @@ static HTMLElement* topmostPopoverAncestor(Element& newPopover)
     return topmostAncestor.get();
 }
 
+void HTMLElement::checkAndPossiblyClosePopoverStackInternal()
+{
+    Vector<RefPtr<Element>> autoPopoverList;
+    for (auto& element : document().topLayerElements()) {
+        if (!is<HTMLElement>(element) || downcast<HTMLElement>(element.get()).popoverState() != PopoverState::Auto)
+            continue;
+        autoPopoverList.append(element.ptr());
+    }
+
+    for (size_t i = autoPopoverList.size(); i > 1; i--) {
+        if (topmostPopoverAncestor(*autoPopoverList[i - 1]) != autoPopoverList[i - 2]) {
```

Comment:
Can we write this to match the algorithm?

```suggestion
    for (size_t i = autoPopoverList.size() - 1; i > 0; i--) {
        if (topmostPopoverAncestor(*autoPopoverList[i]) != autoPopoverList[i - 1]) {
```

Reply:
Yes, but then i needs to be an int since autoPopoverList.size() - 1 can be negative.

Reply:
Alan suggested this pattern: `for (size_t i = autoPopoverList.size(); i--;)` which should keep `size_t`.

Reply:
@rwlbuis can we follow this pattern?

```
for (size_t i = autoPopoverList.size(); i--;) {
    if (topmostPopoverAncestor(*autoPopoverList[i]) != autoPopoverList[i - 1]) {
```

Reply:
That crashes, probably for the i == 0 case.

Reply:
@alanbaradlay @cdumez wdyt?

Reply:
@rwlbuis This should work:
```
for (size_t i = autoPopoverList.size(); i-- > 1;) {
    if (topmostPopoverAncestor(*autoPopoverList[i]) != autoPopoverList[i - 1]) {
```

Reply:
That works locally, thanks!